sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/UnreserveOfferRecommendation.java

```diff
@@ -16,13 +16,21 @@
 
     public UnreserveOfferRecommendation(Offer offer, Resource resource) {
         this.offer = offer;
+        Resource.Builder resourceBuilder = resource.toBuilder();
+
+        if (resource.hasDisk() && resource.getDisk().hasSource()) {
```

Comment:
Can you include a comment similar to what they have in Marathon to indicate _why_ this is being done?